src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Options/PackageSourcesPage.cs

```diff
@@ -273,5 +308,74 @@ private static ExternalSettingOperationResult<T> GetValuePackageSources<T>(List<
 
             return result;
         }
+
+        public OneOrMany<SettingMessage> ValidateSetting(string moniker, object value)
+        {
+            return default;
+        }
+
+        public OneOrMany<SettingMessage> ValidateArrayItemProperty(string arraySettingMoniker, int arrayItemIndex, string propertyMoniker, IReadOnlyList<IReadOnlyDictionary<string, object>> arraySettingContent)
+        {
+            var settingMessages = new OneOrMany<SettingMessage>();
+            var packageSourceDictionaryList = arraySettingContent as IReadOnlyList<IReadOnlyDictionary<string, object>>;
+            if (packageSourceDictionaryList is null)
+            {
+                throw new InvalidOperationException();
+            }
```

Comment:
```suggestion
            var packageSourceDictionaryList = (IReadOnlyList<IReadOnlyDictionary<string, object>>)arraySettingContent;
```

Invalid casts will give us details about what the actual type is, making investigating the bug a bit easier. It being fewer lines is a nice benefit as well.

Reply:
Great catch, I actually no longer need the cast at all!